src/services/cube-handler.ts

```diff
@@ -452,38 +509,66 @@ export async function createDateDimension(quack: Database, extractor: object | n
 }
 
-export const createLookupTableQuery = (factTableColumn: FactTableColumn) => {
-  return `CREATE TABLE ${makeCubeSafeString(factTableColumn.columnName)}_lookup (
-    "${factTableColumn.columnName}" ${factTableColumn.columnDatatype} NOT NULL,
-    language VARCHAR(5) NOT NULL,
-    description TEXT NOT NULL,
-    notes TEXT,
-    sort_order INTEGER,
-    hierarchy ${factTableColumn.columnDatatype}
-  );`;
+export const createLookupTableQuery = (
+  lookupTableName: string,
+  referenceColumnName: string,
+  referenceColumnType: string
+) => {
+  return pgformat(
+    'CREATE TABLE %I (%I %s NOT NULL, language VARCHAR(5) NOT NULL, description TEXT NOT NULL, notes TEXT, sort_order INTEGER, hierarchy %s);',
+    lookupTableName,
+    referenceColumnName,
+    referenceColumnType,
+    referenceColumnType
+  );
 };
 
-// This is a short version of the validate lookup table code found in the dimension process.
-// This concise version doesn't return any information on why the creation failed.  Just that it failed
-export async function createLookupTableDimension(quack: Database, dataset: Dataset, dimension: Dimension) {
-  logger.debug(`Creating and validating lookup table dimension ${dimension.factTableColumn}`);
-  if (!dimension.lookupTable) return;
-  if (!dimension.extractor) return;
+async function setupLookupTableDimension(
+  quack: Database,
+  dataset: Dataset,
+  dimension: Dimension,
+  viewSelectStatementsMap: Map<Locale, string[]>,
+  rawSelectStatementsMap: Map<Locale, string[]>,
+  joinStatements: string[],
+  orderByStatements: string[]
+) {
   const factTableColumn = dataset.factTable?.find((col) => col.columnName === dimension.factTableColumn);
   if (!factTableColumn) {
     const error = new CubeValidationException(`Fact table column ${dimension.factTableColumn} not found`);
     error.type = CubeValidationType.FactTableColumnMissing;
     error.datasetId = dataset.id;
     throw error;
   }
-  await quack.exec(createLookupTableQuery(factTableColumn));
+  const dimTable = `${makeCubeSafeString(dimension.factTableColumn)}_lookup`;
+  await createLookupTableDimension(quack, dataset, dimension, factTableColumn);
+  SUPPORTED_LOCALES.map((locale) => {
+    const columnName = dimension.metadata.find((info) => info.language === locale)?.name || dimension.factTableColumn;
+    viewSelectStatementsMap.get(locale)?.push(`${dimTable}.description as "${columnName}"`);
+    rawSelectStatementsMap.get(locale)?.push(`${dimTable}.description as "${columnName}"`);
+  });
+  joinStatements.push(
+    `LEFT JOIN "${dimTable}" on "${dimTable}"."${factTableColumn.columnName}"=${FACT_TABLE_NAME}."${factTableColumn.columnName}" AND "${dimTable}".language=#LANG#`
+  );
+  orderByStatements.push(`"${dimTable}".sort_order`);
+}
+
+// This is a short version of the validate lookup table code found in the dimension process.
+// This concise version doesn't return any information on why the creation failed.  Just that it failed
```

Comment:
What's the reason for two versions? Is this faster? If not, what about just using a single version and ignoring/discarding any info you don't need?

Reply:
Old comment... I removed validation a little while ago... Working on the assumption that to get to this stage we've already validated the lookup table. So I'll remove the comment